packages/compiler/core/parser.ts

```diff
@@ -3088,6 +3088,16 @@ function visitEach<T>(cb: NodeCallback<T>, nodes: readonly Node[] | undefined):
  * @param position Position
  * @param filter Filter if wanting to return a parent containing node early.
  */
+export function getNodeAtPosition(
```

Comment:
do we need this overload, wouldn't TypeScript infer `node` if you don't pass `filter`?

Reply:
I might not understand the suggestion, but I was trying to avoid making passing a filter that returns a boolean and is not a type predicate still work so there's no breaking change. If there's a way to have one overload and keep that compat, please let me know how and I'll gladly do it. :)

Reply:
oh I was just wondering if this signature was not covering it all already(can we remove  the non templated one)

```
function getNodeAtPosition<T extends Node>(
  script: TypeSpecScriptNode,
  position: number,
  filter: (node: Node) => node is T
): T | undefined;
```

and we could just have that instead of the overloads but if you say that this caused backward compat issue then I  am probably wrong on what TypeScript can infer

Reply:
I'll double check.

Reply:
Confirming that would be a breaking change, unfortunately:

```
packages/compiler/server/serverlib.ts:1482:56 - error TS2345: Argument of type '(node: Node) => boolean' is not assignable to parameter of type '(node: Node) => node is Node'.
  Signature '(node: Node): boolean' must be a type predicate.

1482   const realNode = getNodeAtPosition(script, position, filter);
                                                            ~~~~~~
```